### DISCUSSION TITLE
Mjolnir integration into GUI

### DISCUSSION THREAD

**Discussion by ghost** [element-hq/element-meta#794] [Ideas]
I think that Element should be able to perform moderation actions using Mjolnir through the GUI, apart from the usual ban/kick/redact. We could include a section to perform those actions with Mjolnir (along with "Deactivate user"), this could even be used to mass redact messages, since that feature is not implemented yet.

EDIT: I'm wondering if there might be a way to integrate moderation bots in general into Element's GUI, without needing to be mjolnir, there are still other moderation bots out there. This might require spec changes since the only real way to do it would probably be by making an standardized API for working with moderation bots, the only problem is that it would be costly and would increase the complexity of the protocol.

**Comment by jimmackenzie** [comment]
Great suggestion!

We should see a reality where Element has first-class moderation tooling built into the clients, and moderation teams in rooms and spaces can layer on moderation tools of their choice to supplement the built-in options, to fit the needs of their communities. That might look like Mjölnir, or using the outcome of the upcoming [Mjölnir for All beta](https://matrix.to/#/#mjolnir-for-all:matrix.org), or other moderation tools.

> EDIT: I'm wondering if there might be a way to integrate moderation bots in general into Element's GUI, without needing to be mjolnir, there are still other moderation bots out there. This might require spec changes since the only real way to do it would probably be by making an standardized API for working with moderation bots, the only problem is that it would be costly and would increase the complexity of the protocol.

We're experimenting with something that could lead to something like this - the [Mjölnir for All beta](https://matrix.to/#/#mjolnir-for-all:matrix.org) will be using Widgets as part of making Mjölnir more widely available. We'll be paying close attention to the feedback through the beta, and using it to inform where we go next in pursuit of first-class, built-in moderation tools for users, room moderators and homeserver admins.

**Reply by ghost** [reply]
@jimmackenzie Thank you! Great comment

**Comment by ghost** [comment]
I am thinking that this could be even used so mjolnir can redact individual messages, that way, people don't get angry when you redact them with your personal account.

**Comment by Konpaku-LVX** [comment]
is Mjolnir like synapse-admin or something?

**Reply by jimmackenzie** [reply]
Mjölnir is a [moderation tool for Matrix](https://github.com/matrix-org/mjolnir).
